File: tests/fixtures/python/simple_functions.py
"""
Simple function examples for testing Python chunking
"""


def add(a: int, b: int) -> int:
    """Calculates the sum of two numbers.

    Args:
        a: First number
        b: Second number

    Returns:
        The sum of a and b
    """
    return a + b


def multiply(a: int, b: int) -> int:
    """Multiplies two numbers."""
    return a * b


def greet(name: str) -> str:
    """Greets a person by name.

    Args:
        name: Person's name

    Returns:
        Greeting message
    """
    return f"Hello, {name}!"


# Function without type hints
def legacy_function(x, y):
    """Legacy function without type hints."""
    return x + y


# Async function
async def fetch_data(url: str) -> str:
    """Fetches data from a URL.

    Args:
        url: URL to fetch from

    Returns:
        Response text
    """
    import aiohttp
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            return await response.text()


# Generator function
def number_generator(max: int):
    """Generates numbers from 0 to max.

    Args:
        max: Maximum number to generate

    Yields:
        Numbers from 0 to max-1
    """
    for i in range(max):
        yield i


# Lambda (anonymous function - should not be chunked)
square = lambda n: n * n


# Function with decorator
def decorator_example(func):
    """Example decorator function."""
    def wrapper(*args, **kwargs):
        print(f"Calling {func.__name__}")
        return func(*args, **kwargs)
    return wrapper


@decorator_example
def decorated_function(x: int) -> int:
    """Function with decorator."""
    return x * 2

File: tests/fixtures/typescript/simple-functions.ts
/**
 * Simple function examples for testing TypeScript chunking
 */

/**
 * Calculates the sum of two numbers
 * @param a First number
 * @param b Second number
 * @returns The sum of a and b
 */
function add(a: number, b: number): number {
  return a + b;
}

/**
 * Multiplies two numbers
 * @param a First number
 * @param b Second number
 * @returns The product of a and b
 */
function multiply(a: number, b: number): number {
  return a * b;
}

/**
 * Greets a person by name
 * @param name Person's name
 * @returns Greeting message
 */
function greet(name: string): string {
  return `Hello, ${name}!`;
}

// Arrow function assigned to const
const square = (n: number): number => {
  return n * n;
};

// Arrow function with expression body
const double = (n: number): number => n * 2;

// Async function
async function fetchData(url: string): Promise<string> {
  const response = await fetch(url);
  return response.text();
}

// Async arrow function
const fetchJson = async (url: string): Promise<unknown> => {
  const response = await fetch(url);
  return response.json();
};

// Generator function
function* numberGenerator(max: number): Generator<number> {
  for (let i = 0; i < max; i++) {
    yield i;
  }
}
